### DISCUSSION TITLE
How to extract the 'fill in/type in' text?

### DISCUSSION THREAD

**Discussion by volunteers** [pymupdf/PyMuPDF#1874] [Looking for help] [answered]
The PDF file has some empty space to let users fill in their names, mailing addresses, etc. How to extract those filled-in texts?
I tried `page.get_text('text')`. It returns all the text except the filled-in text. It seems `page.annots` is not the correct way either. Thanks.

**Comment by JorjMcKie** [comment]
You probably talk about PDF fields. Extract field values of a page like that:
```python
for widget in page.widgets():
    print(widget.field_value)
```

**Reply by volunteers** [reply]
Thanks for the reply.
It is not the PDF fields and I got an empty value with `page.widgets()`. It is a plain PDF and I used Acrobat "Add Text" to insert some texts. Please take a look at the "Name" and "Date" sections on the attached file.  I would like to extract "John Wall" and "8/14/2010".  I tried `page.get_text('text') `and it will not return the desired texts.
[example PDF file](https://drive.google.com/file/d/1UyVseQ-EDY9aD7o_M0bCVJ60zXF4Hjlb/view?usp=sharing)

**Comment by JorjMcKie** [comment, marked as answer]
Well, I don't know what you really did, but this worked for me (look at the bottom):
```python
print(page.get_text())
11
Practice
Name
Date
McGraw-Hill School Division
10
At Home: Work with a family member to write three
complete sentences about how to grow flowers.
McGraw-Hill Language Arts
Grade 3, Unit 1, Sentences, 
pages 2–3
1
Sentences
A. Write each group of words that is a sentence.
1. Places the flowerpot in the sun.
2. Todd waters the plant every day.
3. A green stem grows in the soil.
4. Tiny leaves begin to sprout.
5. Grows bigger and bigger.
B. Write each group of words so that they form a sentence.
6. In the garden, Juan.
7. Finds an empty flowerpot.
8. Plants some seeds.
9. His little sister Rosa.
10. Every day Juan and Rosa.
1
Practice
Name
Date
• A sentence tells a complete thought.
Juan opens a package of seeds.
• A sentence fragment does not tell
a complete thought.
Plants them in a flowerpot.
• Begin every sentence with a capital letter.
REMEMBER THE RULES
John Wall  #<==
8/14/2010  #<==

```
Your next question surely will be: "Why is text that I read at top of page extracted as the last thing?"
Answer: "Because it was entered **_after_** everything else already was there. PDF always does it this way - any other approach will be wrong somewhere in the world, just think of right-to-left scripts or bottom-down."

You can also extract the text sorted from top-left to bottom-right like this, and you see your text at a place you more or less expect.
```python
print(page.get_text(sort=True))
John Wall
8/14/2010
1
Practice
Name
Date
11
Practice
Name
Date
Sentences
REMEMBER THE RULES
• A sentence tells a complete thought.
Juan opens a package of seeds.
• A sentence fragment does not tell
a complete thought.
Plants them in a flowerpot.
• Begin every sentence with a capital letter.
A. Write each group of words that is a sentence.
1. Places the flowerpot in the sun.
2. Todd waters the plant every day.
3. A green stem grows in the soil.
4. Tiny leaves begin to sprout.
5. Grows bigger and bigger.
B. Write each group of words so that they form a sentence.
6. In the garden, Juan.
7. Finds an empty flowerpot.
8. Plants some seeds.
9. His little sister Rosa.
McGraw-Hill School Division
10. Every day Juan and Rosa.
10
At Home: Work with a family member to write three
complete sentences about how to grow flowers.
McGraw-Hill Language Arts
Grade 3, Unit 1, Sentences, 
pages 2–3
1
```

**Reply by volunteers** [reply]
Thanks for the answer. I only checked the top of the texts and didn't notice the filled-in texts were located at the very end.
